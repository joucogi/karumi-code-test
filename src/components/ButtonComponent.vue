<template>
  <div>
    <b-button :id="id" size="lg" variant="primary" @click="$emit('click')">
      <span> {{ name }} </span>
    </b-button>
  </div>
</template>
<script>
export default {
  props: {
    id: {
      type: String,
      default: "btnId",
      required: false
    },
    name: {
      type: String,
      default: "Send",
      required: false
    }
  }
};
</script>
<style scoped>
div {
  text-align: left;
}
span {
  font-weight: bold;
}
</style>
