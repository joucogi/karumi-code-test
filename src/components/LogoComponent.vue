<template>
  <div>
    <img alt="Karumi logo" src="../assets/logo.png" />
  </div>
</template>
<style scoped>
  img {
    width: 60%;
    min-width: 80px;
  }
</style>