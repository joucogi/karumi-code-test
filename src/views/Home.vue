<template>
  <div class="home">
    <div class="logo">
      <LogoComponent />
    </div>
    <div class="content">
      <h2 class="mb-5">Welcome to your homepage {{ name }}</h2>
      <ButtonComponent id="btnLogout" name="Log out" @click="goToLogin" />
    </div>
  </div>
</template>

<script>
import { mapState, mapGetters, mapMutations } from "vuex";
import LogoComponent from "@/components/LogoComponent";
import ButtonComponent from "@/components/ButtonComponent";
import { LOGOUT } from "@/store/mutations";
import { USER_IS_LOGGED } from "@/store/getters";

export default {
  name: "Home",
  components: {
    LogoComponent,
    ButtonComponent
  },
  computed: {
    ...mapState(["user"]),
    ...mapGetters({
      userIsLoggedIn: USER_IS_LOGGED
    }),
    name: function() {
      return this.userIsLoggedIn ? this.user.name : "Unknown";
    }
  },
  methods: {
    ...mapMutations({
      logout: LOGOUT
    }),
    goToLogin: function() {
      this.logout();
      this.$router.push({ name: "Login" });
    }
  }
};
</script>
<style scoped>
.home {
  width: 100%;
  max-width: 500px;
  display: flex;
  flex-direction: column;
}

.home > div {
  flex: 1;
}

.logo {
  display: flex;
  align-items: center;
  justify-content: center;
  align-content: center;
}

.content {
  display: flex;
  flex-direction: column;
  padding: 50px;
}
</style>
