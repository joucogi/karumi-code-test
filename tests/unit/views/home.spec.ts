import { shallowMount, Wrapper } from "@vue/test-utils";
import Vue from "vue";
import Vuex from "vuex";

import Home from "@/views/Home.vue";
import LogoComponent from "@/components/LogoComponent.vue";
import ButtonComponent from "@/components/ButtonComponent.vue";
import {
  toMatchSnapshot,
  componentExists,
  createEmptyStore,
  createStoreWithUser
} from "../helpers";
import State from "@/store/state";

Vue.use(Vuex);

let wrapper: Wrapper<Vue>;

beforeEach(() => {
  const store = createEmptyStore(new State());
  wrapper = shallowMount(Home, {
    store
  });
});

describe("Home View", () => {
  it("contains a LogoComponent", () => {
    componentExists(wrapper, LogoComponent);
  });

  it("contains an H2 tag with welcome message when user is not logged in", () => {
    const h2 = wrapper.find("h2");
    expect(h2.text()).toContain("Welcome to your homepage Unknown");
  });

  it("contains an H2 tag with welcome message when user is logged in", () => {
    const store = createStoreWithUser();
    wrapper = shallowMount(Home, {
      store
    });

    const h2 = wrapper.find("h2");
    expect(h2.text()).toContain(`Welcome to your homepage ${name}`);
  });

  it("contains a ButtonComponent", () => {
    componentExists(wrapper, ButtonComponent);
  });

  it("renders correctly when user is not logged in", () => {
    toMatchSnapshot(wrapper);
  });

  it("renders correctly when user is logged in", () => {
    const store = createStoreWithUser();
    wrapper = shallowMount(Home, {
      store
    });

    toMatchSnapshot(wrapper);
  });
});
