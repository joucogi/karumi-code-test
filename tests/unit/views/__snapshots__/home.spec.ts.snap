// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home View renders correctly when user is logged in 1`] = `
<div
  class="home"
>
  <div
    class="logo"
  >
    <logocomponent-stub />
  </div>
   
  <div
    class="content"
  >
    <h2
      class="mb-5"
    >
      Welcome to your homepage Joel
    </h2>
     
    <buttoncomponent-stub
      id="btnLogout"
      name="Log out"
    />
  </div>
</div>
`;

exports[`Home View renders correctly when user is not logged in 1`] = `
<div
  class="home"
>
  <div
    class="logo"
  >
    <logocomponent-stub />
  </div>
   
  <div
    class="content"
  >
    <h2
      class="mb-5"
    >
      Welcome to your homepage Unknown
    </h2>
     
    <buttoncomponent-stub
      id="btnLogout"
      name="Log out"
    />
  </div>
</div>
`;
